### DISCUSSION TITLE
nestjs中service与microservice有什么差别

### DISCUSSION THREAD

**Discussion by PizzaJu** [nestcn/docs.nestjs.cn#268] [2-Q&A]
在一个单体项目中，如果不使用microservice，同样可以使用注入的方式，对service中的方法进行调用。microservice除了支持一些消息介质(如mq、rpc等)外，看起来与普通service的作用一样。那么两者之间的本质差别是什么？

**Comment by ganshiqingyuan** [comment]
本质是一个挂了不影响另一个啊，，微服务的优点，，

**Comment by PizzaJu** [comment]
可能我的打开方式不对，在一个项目里去验证demo了
但很疑惑如果是不同的项目之间，为什么在官网文档里没有看到不同端口或者不同IP的定义，难道只通过引入相同名字的客户端就能通信了吗？
![Uploading image.png…]()

**Comment by PizzaJu** [comment]
文档里还是有定义host和port的地方的

![image](https://user-images.githubusercontent.com/32564579/129011443-f34b9d33-8460-45ca-a27b-d7afcd9bf70a.png)

只是代码例子中没有采用的是默认host为localhost，默认port为3000。可以按照官网例子定义两个项目通过微服务进行通讯，过程中关闭接受方，再进行测试会发现提示“connect ECONNREFUSED 127.0.0.1:3000”
